xc/xenops_server_xen.ml

```diff
@@ -1581,6 +1581,27 @@ module VM = struct
 									(fun fd ->
 										Domain.restore task ~xc ~xs ~store_domid ~console_domid ~no_incr_generationid (* XXX progress_callback *) ~timeoffset ~extras build_info (choose_xenguest vm.Vm.platformdata) domid fd
 									);
+								if inject_igmp_query then
+									try
+										let get_vif_interface_name vif =
+											let get_frontend_domid_of_vif vif =
+												let vmid = fst vif.Vif.id in
+												let domid = match (with_xc_and_xs (fun xc xs -> domid_of_uuid ~xc ~xs Newest (uuid_of_string vmid))) with
+													| Some x -> x
+													| None -> raise (Does_not_exist ("domid_of_uuid", vmid))
+												in
+												domid
+											in
+											let domid = get_frontend_domid_of_vif vif in
+											Printf.sprintf "vif%d.%s" domid (snd vif.Vif.id)
+										in
+										let script = "/usr/libexec/xenopsd/igmp_query_injector.py" in
```

Comment:
Please don't hardcode the path, but add it to xc_resource.ml, so that it can be set from the config file.